src/openrct2-ui/windows/TileInspector.cpp

```diff
@@ -712,6 +711,12 @@ class TileInspector final : public Window
 
     void OnClose() override
     {
+        ToolCancel();
+        auto elem = GetSelectedElement();
+        if (elem != nullptr)
+        {
+            MapInvalidateElement(_toolMap, elem);
```

Comment:
This is a bit problematic in multiplayer, someone else might have deleted the tile element. Its probably better to store the position of the tile where the element was selected and use that to invalidate the tile, the pointer is only used during the rendering path to see if it requires to be highlighted without accessing the actual memory of it.

Reply:
That's already what this code does. `GetSelectedElement()` uses the position of the selected element combined with the selected index to return a pointer.

Reply:
On second thought, the selected index could be greater than the number of elements in the tile if someone deletes stuff on multiplayer. I'm not fully sure, but I think this would break everything that uses the tile inspector selection if it doesn't listen to updates. 

The only solution would be to re-iterate through the tile and count the number of elements every time the selection is used.

Reply:
We don't need to know which specific element was selected for invalidation to work, the invalidation is actually quite primitive as it just marks on a fixed size grid which area of the screen requires to be redrawn. Storing a pointer to the tile element to be read at a later point of time is not safe to do.